addons/account/models/account_move.py

```diff
@@ -2002,6 +2002,19 @@ def needed():
                                     ignore = False
                         if ignore:
                             del res[key]
+
+            # Convert float values to their "ORM cache" one to prevent different rounding calculations
+            for dict_key in res:
+                move_id = dict_key.get('move_id')
+                if not move_id:
+                    break
+                record = self.env['account.move'].browse(move_id)
```

Comment:
Why `break` and not `continue`?
```suggestion
                move = self.env['account.move'].browse(dict_key.get('move_id'))
                if not move:
                    continue
```

Reply:
in the test that I did the dictionaries always had the same structure, but indeed for safety let's just skip it